qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -519,6 +520,81 @@ def _check_special_columns(cls, md_template, obj):
             cls._check_template_special_columns(md_template, obj))
 
     @classmethod
+    def _add_common_creation_steps_to_queue(cls, md_template, obj_id,
+                                            conn_handler, queue_name):
+        r"""Adds the common creation steps to the queue in conn_handler
+
+        Parameters
+        ----------
+        md_template : DataFrame
+            The metadata template file contents indexed by sample ids
+        obj_id : int
+            The id of the object being created
+        conn_handler : SQLConnectionHandler
+            The connection handler object connected to the DB
+        queue_name : str
+            The queue where the SQL statements will be added
+        """
+        # Get some useful information from the metadata template
+        sample_ids = md_template.index.tolist()
+        num_samples = len(sample_ids)
+        headers = list(md_template.keys())
+
+        # Get the required columns from the DB
+        db_cols = get_table_cols(cls._table, conn_handler)
```

Comment:
I was thinking on that, but this is a private function and it should be only used in the `create` functions, that already have that test. If you feel strong about it, I can add it though...